README.md

```diff
@@ -108,43 +108,92 @@ See [CONFIG.md](CONFIG.md) for details.
 See [Code Samples](https://github.com/hazelcast/hazelcast-nodejs-client/tree/master/code_samples)
 
 
-# Serialization Considerations
+# Serialization
+**Table of Contents**
+* [1. IdentifiedDataSerializable Serialization](#1-identifieddataserializable-serialization)
+* [2. Portable Serialization](#2-portable-serialization)
+* [3. Custom Serialization](#3-custom-serialization)
+* [4. Global Serialization](#4-global-serialization)
 
-Hazelcast needs to serialize objects in order to be able to keep them in the server memory. For primitive types, it uses Hazelcast native serialization. For other complex types (e.g. JS objects), it uses JSON serialization.
 
-For example, when you try to query your data using predicates, this querying is handled on the server side so Hazelcast does not have to bring all data to the client but only the relevant entries. Otherwise, there would be a lot of unneccessary data traffic between the client and the server and the performance would severely drop.
-Because predicates run on the server side, the server should be able to reason about your objects. That is why you need to implement serialization on the server side.
+Hazelcast serializes all your objects into an instance of `Data`. `Data` is the binary representation of an object. When Hazelcast serializes an object into `Data`, it first checks whether the object is an instance of `IdentifiedDataSerializable`, if not it checks if it is an instance of `Portable` and serializes it accordingly. For the following types Hazelcast optimizes the serialization a user can not override this behaviour.
```

Comment:
`Data` is an implementation detail. There is no way user sees a `Data` instance in normal operation afaik. It should not be part of the guide.

Reply:
I referenced Java reference manual. See: http://docs.hazelcast.org/docs/3.0/manual/html/ch03.html. I am trying to be as consistent with it as possible.

Reply:
This does not justify putting an implementation detail in the guide. Furthermore, being consistent with the reference manual is an anti-goal. This is a guide targeting people that does not know anything about Hazelcast. Reference manual targets a different audience. The reference manual is Java centric which is again just the opposite of what we are trying to do here.

To me, as a new user trying to understand how I can use Hazelcast and Hazelcast Node.js Client, objects being converted to some internal representation does not mean anything.